libsolidity/ast/Types.cpp

```diff
@@ -889,6 +889,9 @@ tuple<bool, rational> RationalNumberType::isValidLiteral(Literal const& _literal
 		case Literal::SubDenomination::Wei:
 		case Literal::SubDenomination::Second:
 			break;
+		case Literal::SubDenomination::Gwei:
+			value *= bigint(1'000'000'000);
```

Comment:
What is the type of `1000000000`? Why the conversion to bigint? I think a string would be much safer.

Reply:
originally it was just a string, then reviewers wanted the digit separators which only works as an number literal (at least I didn't find any for the multiprecision boost class) so I changed it to a string. I suppose the conversion was just a left-over from the change from string to int literal.

I'll change it back to a string